doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -1,102 +1,120 @@
-=================================
-Term rewriting and simplification
-=================================
+=========================
+Reasoning with equalities
+=========================
 
-.. _rewritingexpressions:
+There are multiple notions of :gdef:`equality`:
 
-Rewriting expressions
----------------------
+- :gdef:`Liebniz equality` is the standard
+  way to define equality in Coq and the Calculus of Inductive Constructions, which is in
+  terms of a relation.  The standard library defines `eq` this way:
 
-These tactics use the equality :g:`eq:forall A:Type, A->A->Prop` defined in
-file ``Logic.v`` (see :ref:`coq-library-logic`). The notation for :g:`eq T t u` is
-simply :g:`t=u` dropping the implicit type of :g:`t` and :g:`u`.
+   Inductive eq (A:Type) (x:A) : A -> Prop := eq_refl : x = x
 
-.. tacn:: rewrite @term
-   :name: rewrite
+   .. todo PR: the definition in .v file is a little different
 
-   This tactic applies to any goal. The type of :token:`term` must have the form
+- :gdef:`Setoid equality <setoid equality>` defines equality in terms of an equivalence
+  relation.  A :gdef:`setoid` is a set that is equipped with an equivalence relation
+  (see https://en.wikipedia.org/wiki/Setoid).  These are needed to form a :gdef:`quotient set`
+  or :gdef:`quotient`
+  (see https://en.wikipedia.org/wiki/Equivalence_Class).  In Coq, users generally work
+  with setoids rather than constructing quotients, for which there is no specific support
+  (see also `#10871 <https://github.com/coq/coq/issues/10871>`_).
 
-   ``forall (x``:sub:`1` ``:A``:sub:`1` ``) ... (x``:sub:`n` ``:A``:sub:`n` ``), eq term``:sub:`1` ``term``:sub:`2` ``.``
+- :gdef:`Definitional equality <definitional equality>` is equality based on the conversion rules, which
+  Coq can determine automatically.  When two terms are definitionally equal, Coq knows it can
+  replace one with the other, such as with :tacn:`change` `X with Y`, among many
+  other advantages.
 
-   where :g:`eq` is the Leibniz equality or a registered setoid equality.
+  .. todo PR: A and B are convertible iff they are definitionally equal??
```

Comment:
Yes. And some error messages use the word "not convertible" instead of "not definitionally equal" BTW.